enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,756 @@
+This is also a concept in Cluster API.
+When converting ControlPlaneMachineSets to Cluster API, we will ensure that the appropriate failure domain configuration is configured inside the Cluster API InfrastructureCluster,
+and that the failure domains from the ControlPlaneMachineSet Machine API configuration are reflected across.
+
+When deploying Machines using the future Cluster API extension, the InfrastructureTemplate will be combined with the named failure domains on the InfrastructureCluster
+to spread the Machines across multiple failure domains.
+
+#### Creation of new Machines
+
+When a Machine API MachineSet creates a new Machine, the synchronisation controller will create the equivalent Cluster API Machine based on the method described in [Machine Synchronisation](#machine-synchronisation).
+
+When a Cluster API MachineSet creates a new Machine, the synchronisation controller will create it’s Machine API equivalent if, and only if, the Cluster API MachineSet has a Machine API equivalent.
+This will allow users to switch a MachineSet back to Machine API at a later date if they wish to do so.
+
+When the Cluster API MachineSet has been created independently of Machine API, new Machines will not be created in Machine API to reflect the Machines owned by the Cluster API MachineSet.
+
+#### Deletion of Machines
+
+To enable the synchronisation of deletion, the synchronisation controller will use its own Finalizer on resources that have Machine API and Cluster API equivalents.
+
+This means that for a Machine, we expect the authoritative controller to perform the actual deletion work within the infrastructure provider.
+Once it has completed this effort, it will remove its own Finalizer.
+
+As soon as the synchronisation controller notices the Machine has been deleted, it should ensure that the Machine API/Cluster API equivalent (if present) is also deleted.
+It should then wait until the authoritative Machine controller removes its finalizer before removing both its own, and the Machine API/Cluster API Machine finalizer if present.
+In scenarios where both Machine API and Cluster API have been authoritative, it is expected that both Machine controllers will have added their own Finalizers, therefore we expect the synchronisation controller should be able to handle this.
+
+It is also feasible that a customer may want to remove the Machine API resources after they have migrated to Cluster API and no longer require the Machine API synchronisation.
+To allow for this, if a non-authoritative Machine API parent[^8] resource is deleted, the deletion event will not be synchronised to the Cluster API equivalent and the synchronisation controller will ensure finalizers are removed as appropriate.
+In this scenario, the Cluster API equivalent will continue to operate on its own.
+To ensure users are aware of this situation, webhooks will be introduced to warn users that the deletion will not be synchronised.
```

Comment:
Yes there is a warning type. For a few versions of kube now you can return warning messages to users instead of errors. With that, the operation still succeeds, but, kubectl or the UI or whatever they are using should print out a message on their display to give them the content of the warning you sent.

Reply:
TIL